src/Elastic.Apm/Metrics/MetricsProvider/ProcessTotalCpuTimeProvider.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Elastic.Apm.Api;
+
+namespace Elastic.Apm.Metrics.MetricsProvider
+{
+	public class ProcessTotalCpuTimeProvider : IMetricsProvider
+	{
+		private const string ProcessCpuTotalPct = "system.process.cpu.total.norm.pct";
+		private readonly object _lock = new object();
+		private bool _first = true;
+		private TimeSpan _lastCurrentProcessCpuTime;
+		private DateTime _lastTick;
+
+		private Version _processAssemblyVersion;
+
+		public int ConsecutiveNumberOfFailedReads { get; set; }
+		public string NameInLogs => "total process CPU time";
```

Comment:
Changed it. Mainly because this is in my opinion more readable. I'm always confused by the keys we use `system.process.cpu.total.norm.pct`, then the terms used in windows vs. linux word. I think it best to describe it in a readable way.

Reply:
By class name I meant .NET class name (in this case it would be "ProcessTotalCpuTimeProvider") but I'm okay with either approach as long as it's consistent.